ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -205,6 +212,33 @@ PullQueryResult executePullQuery(
           Optional.ofNullable(finalSourceType).orElse(PullSourceType.UNKNOWN),
           Optional.ofNullable(finalPlanType).orElse(PullPhysicalPlanType.UNKNOWN),
           Optional.ofNullable(finalRoutingNodeType).orElse(RoutingNodeType.UNKNOWN)));
+
+      final String stmtLower = statement.getStatementText().toLowerCase(Locale.ROOT);
+      final String messageLower = e.getMessage().toLowerCase(Locale.ROOT);
+      final String stackLower = Throwables.getStackTraceAsString(e).toLowerCase(Locale.ROOT);
+
+      // do not include the statement text in the default logs as it may contain sensitive
+      // information - the exception which is returned to the user below will contain
+      // the contents of the query
+      if (messageLower.contains(stmtLower) || stackLower.contains(stmtLower)) {
+        final StackTraceElement loc = Iterables
+            .getLast(Throwables.getCausalChain(e))
+            .getStackTrace()[0];
+        LOG.error("Failure to execute pull query {} {}, not logging the error message since it "
+            + "contains the query string, which may contain sensitive information. If you "
+            + "see this LOG message, please submit a GitHub ticket and we will scrub "
+            + "the statement text from the error at {}",
+            routingOptions.debugString(),
+            pullPlannerOptions.debugString(),
+            loc);
+      } else {
+        LOG.warn("Failure to execute pull query. {} {}",
```

Comment:
Why we want to log as WARN v.s. ERROR in the else condition? I feel they are at the same severity level and the only reason for this condition is to be cautious of not printing the query stmt.

Reply:
Most of these errors are no actionable and just require retries, so I assumed they should be logged as `WARN`. The `ERROR` in the if condition is to flag with higher priority that there's something in our code that needs to be fixed. I'm happy logging them both as ERROR or WARN though, not a big deal.

Reply:
Hmm, maybe I missed something important here: I thought the only rationale for differentiating them is to not disclose the query stmt from the error message or stack trace.

Are we assuming that, inside the callees deep-down, we have a standard pattern that if the error is due to our code, then we include include the query stmt in the error message, otherwise we would not??

Reply:
> Hmm, maybe I missed something important here: I thought the only rationale for differentiating them is to not disclose the query stmt from the error message or stack trace.

That was the case - I will go ahead and just make them both `ERROR`.